src/npc.cpp

```diff
@@ -2722,7 +2727,16 @@ bool npc::dispose_item( item& obj, const std::string & )
         }
     }
 
-    if( volume_carried() + obj.volume() <= volume_capacity() ) {
+    //see if it can be stored in inventory
+    npc projected = *this;
+    projected.i_add( obj );
+    if( swap_obj ) {
+        int swap_pos = get_item_position( swap_obj );
+        if( swap_pos != INT_MIN ) {
+            projected.i_rem( swap_pos );
```

Comment:
What if the added item is inserted before the `swap_obj`. The `swap_pos` would not refer to the same item in the `projected` npc.